Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -64,10 +66,24 @@ public RestChannel(RestServiceCollection serviceCollection, string endpoint, Cha
 
         private async Task<HttpResponseMessage> SendAsync<TRequest>(RestMethod restMethod, string host, CallOptions options, TRequest request, CancellationToken cancellationToken)
         {
-            // TODO: Add headers from options.Headers, but intercept the x-goog-api-client header.
+            var headers = options.Headers
+                .Where(mh => !mh.IsBinary)
+                .Where(mh=> mh.Key != VersionHeaderBuilder.HeaderName)
+                .Select(mh => new KeyValuePair<string, string>(mh.Key, mh.Value))
+                .ToList();
+
+            var restVersion = new VersionHeaderBuilder()
```

Comment:
We could presumably construct this once and store it as a static variable, right? I think that would be better, as I don't know what the efficiency of these methods is like.

Reply:
👍